### ISSUE TITLE
Failing performance tests on Jenkins

### ISSUE THREAD

**Issue by wojtek-t** [GoogleCloudPlatform/kubernetes#7561]
The first failure happened at ~9:20 PST 29/04/2015. The failure looks like that:

/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/density.go:158
Expected error:
    <*errors.errorString | 0xc20b84daa0>: {
        s: "Controller my-hostname-density3000-75cb549b-ee8e-11e4-ae78-42010af01555: Only found 3052 replicas out of 3000",
    }
    Controller my-hostname-density3000-75cb549b-ee8e-11e4-ae78-42010af01555: Only found 3052 replicas out of 3000
not to have occurred

It seems that because of some reason we are starting much more pods within a replication controller than expected. Another related failure:

Expected error:
    <*errors.errorString | 0xc2083b3050>: {
        s: "Number of reported pods changed: 3222 vs 3000",
    }
    Number of reported pods changed: 3222 vs 3000
not to have occurred

cc @quinton-hoole @lavalamp @bprashanth @fgrzadkowski

**Comment by wojtek-t** [comment]
Feel free to move it outside v1.0 if you can explain it - if not, we should definitely try to understand it.

**Comment by wojtek-t** [comment]
Also note that this is a clear regression - I didn't observe any such failure before. That's why I'm marking it (at least for now) as v1.0 issue.

**Comment by quinton-hoole** [comment]
Might end up being a duplicate of #7548, but let's investigate both in parallel until we confirm.

**Comment by wojtek-t** [comment]
>
Might end up being a duplicate of #7548, but let's investigate both in parallel until we confirm.

I think those are unrelated. Basically, the GCE tests are green now, whereas performance tests didn't pass since then even once.


My feeling is that it might be related to: #6866 @lavalamp @bprashanth - any thoughts about.

**Comment by bprashanth** [comment]
Logs at V(4) would help, that can certainly happen if the watch takes O(minutes) to deliver notifications (which I've seen happen when we, say, delete 1000 pods at once). If that's the problem this essentially boils down to: watch is slow and things timeout because they don't know if a packet was lost or still in flight; to which there are 2 solutions: make watch faster, increase the timeout. 

The good news is that the rc will balance itself out over time as the appropriate notifications get there. So if you really want to track the exact number of replicas at scale minute by minute, its going to be hard. Previously we didn't have this problem because the rcs would poll every 5s, which has various other implications. Currently that 5s poll is replaced by 4 things:
1. watchers on pods and controllers
2. a periodic relist of the rcs every 30s
3. a periodic relist of pods every 5m
4. a ttl on what the rc expects to see via watch (i.e if it created 3 pods it expects to see 3 watch notifications) every 2m

So it feels like we're hitting the 4th case (I can't be sure without logs), and that if we didn't care about the replica count being exact minute to minute, we would be ok after a bit.

**Comment by bprashanth** [comment]
There are at least 2 issues that can lead to a wrong number of replicas at this stage of the test:

1. Slow create->watch cycle (by slow i mean slower than the 2 node-50 pods cluster case). See comments in #7628 for more details.
2. Scheduling pods on dead nodes (https://github.com/GoogleCloudPlatform/kubernetes/issues/7222) or nodes dying after pods have been scheduled on them. This will cause the nodecontroller to delete pods after the rc has observed them to be running, and the test will panic because of the mismatch. 

I'm am still digging into 2, I've seen it happen at least 2/5 runs. In both cases the system will correct itself given time.

**Comment by wojtek-t** [comment]
Thanks for investigating it @bprashanth 
It seems that #7628 fixed huge majority of flakes. For the the fix!
And it seems that the remaining flakes are exactly due to #7222 that you described.

Since the flakes are currently very rarely I'm decreasing the priority of this bug.
Also, I think that we can close this PR as a duplicate of #7222 (feel free to do it).

**Comment by bprashanth** [comment]
There's a third case which is currently the most frequent failure mode:
    <*errors.errorString | 0xc21c4c0580>: {
        s: "Error: Pod my-hostname-density3000-8da6844d-f23a-11e4-bd75-42010af0159ffb5: Container my-hostname-density3000-8da6844d-f23a-11e4-bd75-42010af01555 was found to have terminated 1 times",
    }

It also occasionally crops up in the normal e2es (https://github.com/GoogleCloudPlatform/kubernetes/issues/7548#issuecomment-97964181). Investigating.

**Comment by bprashanth** [comment]
This should fix the unexpected terminations, or reveal another race condition that needs handling: https://github.com/GoogleCloudPlatform/kubernetes/pull/7749

**Comment by wojtek-t** [comment]
@bprashanth thanks for the heads-up.

**Comment by bprashanth** [comment]
The mentioned pr has merged, and now I'm seeing high latency requests of 2 types:
1. Listing events: this doesn't sound like reason to fail the test, can we ignore  them? @wojtek-t @rrati
2. Lising pods: happens less often, investigating.

To be clear I'm talking about the failure message:
```
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/density.go:164
Expected
    <int>: 2
not to be >
    <int>: 0
```

which inevitably end up being:
```
INFO: WARNING - requests with too high latency: {verb:LIST resource:events quantile:0.9 latency:11844619000}
INFO: WARNING - requests with too high latency: {verb:LIST resource:events quantile:0.99 latency:14242416000}
```

**Comment by wojtek-t** [comment]
Correct - this is something we're currently looking at.
cc @fgrzadkowski

**Comment by quinton-hoole** [comment]
The basic event listing e2e test is intermittently failing in the same way as bprashanth@ mentions above, so it's failure is currently somewhat independent of scale.   I'd be very happy to temporarily disable that part of this scalability test until the flakyness of the basic event listing test has been deflaked.

To be clear, I'm talking about this flaky basic event listing e2e test:

==== snip snip from out continuous integration tests =============
	
Identified problems

Events should be sent by kubelets and the scheduler about pods scheduling and running

/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/events.go:124
kubelet events from running pod
Expected
    <int>: 0
not to be zero-valued

**Comment by fgrzadkowski** [comment]
@bprashanth I'll send a PR later today to ignore events related metrics.

I was also investigating slow listing of pods. Main reason currently is GOMAXPROCS set for etcd to 1. Increasing it to NumCores() improves a lot, but metrics are still above 1.0 targets. I believe that now the bottleneck is somewhere in apiserver itself.
Please let know if you find something, so that we don't investigate the same thing :)

**Comment by bprashanth** [comment]
Thanks for the heads up, will keep you posted

**Comment by wojtek-t** [comment]
I've seen another failure of the first type:
```
Density [Performance suite] should allow starting 30 pods per node
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/density.go:164
Expected error:
    <*errors.errorString | 0xc20d608f10>: {
        s: "Number of reported pods changed: 3001 vs 3000",
    }
    Number of reported pods changed: 3001 vs 3000
not to have occurred
```

@bprashanth Do you think that #7869 can help for that?

**Comment by bprashanth** [comment]
Maybe, but it looks like the controller finished creating 3000 pods and the test moved to the stage where we started checking for pod phase. After that something happened that caused another pod creation, so this might be something else. Will try and repro.

**Comment by bprashanth** [comment]
@wojtek-t I think the number of reported pods changing issue is:
https://github.com/GoogleCloudPlatform/kubernetes/issues/7815#issuecomment-100001495

It's certainly possible to have > number of replica pods in the system, if some of them have fallen into an inactive state. I hope to add some clarity with https://github.com/GoogleCloudPlatform/kubernetes/pull/7945.

**Comment by wojtek-t** [comment]
Thanks @bprashanth - will take a look into it.

**Comment by bprashanth** [comment]
The main cause of failure remains the slow list. See: https://github.com/GoogleCloudPlatform/kubernetes/issues/8086. 

@fgrzadkowski do you think listing is the bottleneck right now? (I'm not 100% convinced, but you have more experience in this area). If so, there might be 2 things we can do:

1. I got the latency down from 20ish -> 13ish for the 3000x100 case (raw etcd list was 11ish) by not doing a deep copy when we pull objects out of the cache. For the specific case of list, a deep copy *shouldn't* be necessary. There are however other callers of that decoding routine that might need the copy. And of course the go race detector.

2. The latencies go up when all kubelets hit the apiserver at once, which probably happens today because of https://github.com/GoogleCloudPlatform/kubernetes/issues/6770

**Comment by wojtek-t** [comment]
>
I got the latency down from 20ish seconds-> 13s for the 3000x100 case (raw etcd list was 11s, with a single core) by not doing a deep copy when we pull objects out of the cache. For the specific case of list, a deep copy shouldn't be necessary. There are however other callers of that decoding routine that might need the copy. And of course the go race detector.

I will be working on optimizing DeepCopy starting today.

**Comment by fgrzadkowski** [comment]
Today I was looking into ugorji JSON codec (https://github.com/ugorji/go/tree/master/codec) to improve some operations in etcd client library. During bechmarking I've realized that we could use the same codec to implement fast DeepCopy by simply encoding to bytes and decoding to new object. My initial benchmark shows ~6x improvement, which is very promising! I'll try to send a PR for this tomorrow.

**Comment by bprashanth** [comment]
With: #8260 I noticed an inverse correlation between number of active watches and list latency. Specifically when the number of open watches in the cluster goes from 500 -> 300 the latency of a list crosses a second (and often goes much beyond).

I also noticed a correlation between number of POSTs and list latency.

Will report back with how this changes when i randomize the timeout as described in https://github.com/GoogleCloudPlatform/kubernetes/issues/6770 (randomizing the timeout is cheap enough that we should be able to do that and batch lists, or whatever fancy solution we might think of).

**Comment by lavalamp** [comment]
> Specifically when the number of open watches in the cluster goes from 500 -> 300 the latency of a list crosses a second (and often goes much beyond).

Isn't the causality here that everyone doing a watch switches to doing a list when the watch ends? It's not that low # of watches directly causes high list latency-- you can't fix this problem by adding even more watches :)

Or do I misunderstand?

**Comment by bprashanth** [comment]
@lavalamp thats exactly it, and the POST stuff that I'm still digging up

**Comment by bprashanth** [comment]
> you can't fix this problem by adding even more watches
what did you mean by that?

**Comment by fgrzadkowski** [comment]
Another solution is to limit number of concurrent list pod requests in apiserver. We already have all the tools for that. I did a simple test locally and it improved latency significantly. I still need to measure how much exactly.

**Comment by bprashanth** [comment]
Of course, more fancy solutions exist such as batch up list requests at peak load for 1/2 second and do a single list etc. The timeout solution is mutually exclusive. @fgrzadkowski by limit do you mean return a dead 500 and assume the clients will retry?

**Comment by fgrzadkowski** [comment]
I mean to reuse this:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/apiserver/handlers.go#L73
but with a different regexp. This helped me reduce CPU usage on apiserver. I can you a PR tomorrow, since I already have it hacked locally.

**Comment by bprashanth** [comment]
@fgrzadkowski It isn't just the concurrent lists, etcd seems to be bad at handling write load as well (that was the correlation between POSTs I mentioned in https://github.com/GoogleCloudPlatform/kubernetes/issues/7561#issuecomment-102116408). Each pod posts at least 7 events on startup (4 for the pod container, 3 for the infra container). With the scheduler handing out 10pps, there are very often 100s of in flight event POST requests hogging apiserver time. The problem here is that we only rate limit the initiation of the request, so a 5qps connections ends up being a sham because that kubelet is already waiting on some 10 event posts. 

Server side rejection should be the last line of defense because we have no choice but to 500. I have a local change that I'm testing out that *also* caps max in flight requests client side. Will report back with results.

**Comment by wojtek-t** [comment]
I think that currently we have 2 biggest problems:

- [ ] tests are flaky in the phase of cleaning up the ReplicationController - DeleteRC is returning an error "timed out waiting for the condition"
- [ ] cluster is not starting due to not created firewall rules:

    ERROR: (gcloud.compute.firewall-rules.create) Some requests did not succeed:
    Did not insert the following resources within 660s:    https://www.googleapis.com/compute/v1/projects/kubernetes-jenkins/global/firewalls/e2e-scalability-minion-e2e-scalability-http-alt.


 I will look into the first issue today. FYI: @fgrzadkowski

**Comment by wojtek-t** [comment]
BTW - the second issue seems to be a GCE issue not the k8s issue.

**Comment by wojtek-t** [comment]
There is one more issue (it happened twice during last day):
- [ ] cluster is not starting due to not all nodes being registered in the cluster:

    [0;31mDetected 99 ready nodes, found 99 nodes out of expected 100. Your cluster may not be working.

**Comment by wojtek-t** [comment]
Regarding cleaning up ReplicationController - #8779 is a workaround (this test should no longer fail on Jenkins).

However, there seem to be some real issue in ReplicationController. Basically looking on the logs we have:

    I0526 08:35:15.796503       5 replication_controller.go:320] Too many "e2e-tests-density-6b911472-c3e2-45b6-9eaf-23d5bbb67c58"/"my-hostname-density3000-fff37f7c-0380-11e5-9570-a0481cabf39b" replicas, need 0, deleting 500
    I0526 08:35:39.339995       5 replication_controller.go:320] Too many "e2e-tests-density-6b911472-c3e2-45b6-9eaf-23d5bbb67c58"/"my-hostname-density3000-fff37f7c-0380-11e5-9570-a0481cabf39b" replicas, need 0, deleting 500
    E0526 08:35:39.528616       5 reflector.go:168] watch of *api.Secret ended with: 401: The event in requested index is outdated and cleared (the requested history has been cleared [14498367/14459192]) [14499366]
    I0526 08:36:05.510472       5 replication_controller.go:320] Too many "e2e-tests-density-6b911472-c3e2-45b6-9eaf-23d5bbb67c58"/"my-hostname-density3000-fff37f7c-0380-11e5-9570-a0481cabf39b" replicas, need 0, deleting 500
    I0526 08:36:31.846577       5 replication_controller.go:320] Too many "e2e-tests-density-6b911472-c3e2-45b6-9eaf-23d5bbb67c58"/"my-hostname-density3000-fff37f7c-0380-11e5-9570-a0481cabf39b" replicas, need 0, deleting 500
    I0526 08:36:57.400387       5 replication_controller.go:320] Too many "e2e-tests-density-6b911472-c3e2-45b6-9eaf-23d5bbb67c58"/"my-hostname-density3000-fff37f7c-0380-11e5-9570-a0481cabf39b" replicas, need 0, deleting 500
    ...
    I0526 08:40:14.139313       5 replication_controller.go:320] Too many "e2e-tests-density-6b911472-c3e2-45b6-9eaf-23d5bbb67c58"/"my-hostname-density3000-fff37f7c-0380-11e5-9570-a0481cabf39b" replicas, need 0, deleting 500

 So basically, ReplicationController manager is not doing anything useful for more than 3 minutes. However, looking at the logs from apiserver pod number 2500 is deleted at 08:37:22:

    ... 2499 deletions of pods within our RC
    I0526 08:37:22.861456       5 handlers.go:135] DELETE /api/v1beta3/namespaces/e2e-tests-density-6b911472-c3e2-45b6-9eaf-23d5bbb67c58/pods/my-hostname-density3000-fff37f7c-0380-11e5-9570-a0481cabf3wiyzn: (33.12162ms) 200 [[kube-controller-manager/v0.17.1 (linux/amd64) kubernetes/bf4cb33] 127.0.0.1:42803]
    I0526 08:40:14.173762       5 handlers.go:135] DELETE /api/v1beta3/namespaces/e2e-tests-density-6b911472-c3e2-45b6-9eaf-23d5bbb67c58/pods/my-hostname-density3000-fff37f7c-0380-11e5-9570-a0481cabf3phb0o: (31.600239ms) 200 [[kube-controller-manager/v0.17.1 (linux/amd64) kubernetes/bf4cb33] 127.0.0.1:43191]
    ... 499 deletions of pods within our RC

It seems like a bug somewhere around replication_controller.go (or in watch?).

FYI @bprashanth @lavalamp @fgrzadkowski

**Comment by wojtek-t** [comment]
I think I understand the issue. What's happening is:
  1. replication_controller.go is trying to remove 500 pods;
  2. it waits until all of them are removed;
  3. in the meantime full resync is happening in the underlying reflector;
  4. this triggers Replace() method in delta_fifo.go; however, that list doesn't contain the pods that are already removed, but those weren't yet delivered (via) watch to the reflector underlying storage (which is delta_fifo.go)
  5. as a result, for those pods, we put DeletedFinalStateUnknown:
https://github.com/GoogleCloudPlatform/kubernetes/blob/4a6ab7fd3698c5cc7172c5bd63efab320d1f2d17/pkg/client/cache/delta_fifo.go#L337
  6. When the DeletedFinalStateUnknown is processed by replication_controller.go, it can't defer what is its corresponding RC and thus cannot update its expectations (here the "obj" is NOT a pod):
https://github.com/GoogleCloudPlatform/kubernetes/blob/ddbf851b6e41df676a5374a66ffc82057f24c2ce/pkg/controller/replication_controller.go#L240
  7. Thus the expectations for the RC will never drop to 0, until the full resync is done (which happens after 3 minutes from the try to remove 500 pods).

 To summarize, from the correctness point of view, it will eventually do its job, but I think it's not the best experience. 

Possible solutions:
  1. Add a cache in replication_controller.go, which will be storing "pod.Name -> its RC" mapping. Thanks to it, we will not need the whole object to determine its RC in deletePod:
https://github.com/GoogleCloudPlatform/kubernetes/blob/ddbf851b6e41df676a5374a66ffc82057f24c2ce/pkg/controller/replication_controller.go#L240
  2. Somehow fix the DeltaFIFO::Replace method. A possible solution would be to force the underlying KeyLister interface to also expose Get() method - then instead of queueing DeletedFinalStateUnknown, queue the result of appropriate Get() method (i.e. the latest known state):
https://github.com/GoogleCloudPlatform/kubernetes/blob/4a6ab7fd3698c5cc7172c5bd63efab320d1f2d17/pkg/client/cache/delta_fifo.go#L337
  3. Leave as is - however I'm really against this, because it can "freeze" ReplicationManager for 3 minutes with respect to some ReplicationController, which isn't a good experience.

@lavalamp @bprashanth - thoughts?
BTW, I'm happy to fix this once we agree on the approach here.

**Comment by bprashanth** [comment]
@wojtek-t we're hitting the case described in https://github.com/GoogleCloudPlatform/kubernetes/issues/8676 from time to time. I already have a pr locally that does 2, just need some unittests. Your conclusion is correct, it will eventually size down but we can be faster if the delta fifo gives us the deleted pod. BTW I'm glad you were able to debug that far in :)

**Comment by wojtek-t** [comment]
@bprashanth - great! What is the expected ETA for you PR?

**Comment by bprashanth** [comment]
@wojtek-t Today

**Comment by wojtek-t** [comment]
@bprashanth - great! - thanks!

**Comment by wojtek-t** [comment]
#8822 is addressing the first issue

**Comment by wojtek-t** [comment]
Also, I already started looking into the third one.

**Comment by wojtek-t** [comment]
I talked with @roberthbailey about the third issue and he said it's reasonable to increase it even to 10 minutes for big (100-node) clusters. Since we're still facing this issue with 5-minute timeout, I'm going to try increasing it to 10 minutes.

**Comment by wojtek-t** [comment]
Regarding firewall, the hypothesis is that it might be related to QPS quota. This might be fixed with #8899

**Comment by wojtek-t** [comment]
It's also possible that nodes node registering themself in the apiserver could be the same issue (lack of QPS quota) - based on what @roberthbailey said.
That's why I'm going to wait for #8899 to be merged before spending more time on it now.

**Comment by wojtek-t** [comment]
To summarize the current state:
- the firewall-related issues were indeed caused by lack of QPS quota - I didn't see any such failure after merging #8899 
- the long running DeleteRC was successfully worked around by #8779 - we still observe it sometimes to take more than 5 minutes, but it doesn't cause test failures now; the proper fix to it #8822 was unfortunately reverted, but there is a separate issue #8676 for

 The only remaining problem that is really hitting us now is:
- [ ] cluster is not starting due to not all nodes being registered in the cluster:

        Detected 99 ready nodes, found 99 nodes out of expected 100. Your cluster may not be working.

**Comment by roberthbailey** [comment]
What is your timeout before failing because the nodes aren't all registered? Are the remaining nodes (1 in your example) NotReady or not registered with the master at all? I don't know that we have done any measurements around how long it takes to assign a CIDR to a kubelet once it has self registered or how long it takes the kubelet to reconfigure cbr0 and mark itself as ready.

**Comment by wojtek-t** [comment]
In all the cases I've seen O(10) - those nodes weren't even registered.

**Comment by wojtek-t** [comment]
Currently, the limit is 5 minutes, but I'm increasing it to 10 minutes now.

**Comment by wojtek-t** [comment]
I was able to catch the above situation on Jenkins and did some more debugging on it. What happened:
- Detected 99 ready nodes, found 99 nodes out of expected 100. Your cluster may not be working.
- I found the node that wasn't registered and after logging to the corresponding VM there is:

        Broken (or in progress) GCE Kubernetes node setup! Suggested first step:
           tail /var/log/startupscript.log

- Looking at the logs from startup script, it basically wasn't able to unpack the salt tree:

        May 29 08:38:01 e2e-scalability-minion-i5x0 startupscript: Unpacking Salt tree
        May 29 08:38:01 e2e-scalability-minion-i5x0 startupscript: 
        May 29 08:38:01 e2e-scalability-minion-i5x0 startupscript: gzip: stdin: not in gzip format
        May 29 08:38:01 e2e-scalability-minion-i5x0 startupscript: tar: Child returned status 1
        May 29 08:38:01 e2e-scalability-minion-i5x0 startupscript: tar: Error is not recoverable: exiting now

- Once looking at the exact file it was trying to unpack we have:

        wojtekt@e2e-scalability-minion-i5x0:/var/cache/kubernetes-install$ cat kubernetes-salt.tar.gz 
        <?xml version='1.0' encoding='UTF-8'?><Error><Code>AccessDenied</Code><Message>Access denied.</Message></Error>wojtekt@e2e-scalability-minion-i5x0:/var/cache/kubernetes-install$ 

- So this seems like some kind of race - I looked in few other machines and they were starting later.

 I'm going to add a mechanism to retry downloading the files if we're not able to unpack them.

**Comment by roberthbailey** [comment]
I guess the good news is that this shouldn't happen on release tars....

This sounds like it can sometimes take a little while for GCS ACLs to propagate (after [this call](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/build/common.sh#L912) is made). I wonder if there is a way to tell GCS that _anything you put into bucket X in the future_ should be automatically marked as world readable.

**Comment by davidopp** [comment]
@wojtek-t is on vacation for the next couple of weeks, so we should either reassign or close if we think #8998 is sufficient. @roberthbailey what do you think?

**Comment by bprashanth** [comment]
When the gce issues are fixed feel free to re-assign to me, I can track the progress of the test on jenkins and follow up on this bug.

**Comment by wojtek-t** [comment]
As of the weekend this wasn't fixed - we had a lot of red runs during the weekend. It seems to be in a better shape now.

 However, there seem to be a huge drop in the performance metrics few hours ago - can someone please take a look at it and rollback the PR that caused the regression?

cc @fgrzadkowski

**Comment by fgrzadkowski** [comment]
@wojtek-t I'm debugging performance regression. I also submitted today #9115 that should fix remaining red runs.

**Comment by fgrzadkowski** [comment]
Majority of e2e scalability jenkins runs are green now. In the last 14 runs we had 12 greens and 2 red ones (failed firewall creating and one due to node being not ready). Let's close this issue and open separate issues if we decide to fix remaining problems.

**Comment by wojtek-t** [comment]
Thanks @fgrzadkowski - SGTM

**Comment by davidopp** [comment]
I'm going to reopen this since there are apparently still problems with e2e scalability test (load test had to be disabled today, #9201). Probably I should open a separate bug but this is close enough.

**Comment by davidopp** [comment]
@fgrzadkowski any update this, or you are still looking into it?

**Comment by fgrzadkowski** [comment]
@davidopp Yes, I'm looking into this. I need to refactor load test to make it more robust.

**Comment by davidopp** [comment]
This is related to #4521.

**Comment by fgrzadkowski** [comment]
In the last runs many of them didn't even start due to failed firewall creation. #9658 should fix this. In the remaining ones most problems were caused by too short timeout for RC creation. Hopefully @9571 should fix it.

I'll watch jenkins closely overnight.
